nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToCSV.java

```diff
@@ -0,0 +1,300 @@
+                }
+            }
+        } else {
+            //the user did not give a list of attributes, take all the attributes from the flowfile
+            result.putAll(ffAttributes);
+        }
+
+        //now glue on the core attributes if the user wants them.
+        if(includeCoreAttributes) {
+            for (String coreAttribute : coreAttributes) {
+                //make sure this coreAttribute is applicable to this flowfile.
+                String val = ff.getAttribute(coreAttribute);
+                if(ffAttributes.containsKey(coreAttribute)) {
+                    if (!StringUtils.isEmpty(val)){
+                        result.put(coreAttribute, val);
+                    } else {
+                        if (nullValForEmptyString) {
+                            result.put(coreAttribute, "null");
+                        } else {
+                            result.put(coreAttribute, "");
+                        }
+                    }
+                }
+            }
+        } else {
+            //remove core attributes since the user does not want them, unless they are in the attribute list. Attribute List always wins
+            for (String coreAttribute : coreAttributes) {
+                //never override user specified attributes, even if the user has selected to exclude core attributes
+                if(!attributes.contains(coreAttribute)) {
+                    result.remove(coreAttribute);
+                }
+            }
+        }
+        return result;
+    }
+
+    private LinkedHashSet<String> attributeListStringToSet(String attributeList) {
+        //take the user specified attribute list string and convert to list of strings.
+        LinkedHashSet<String> result = new LinkedHashSet<>();
+        if (StringUtils.isNotBlank(attributeList)) {
+            String[] ats = attributeList.split(SPLIT_REGEX);
+            for (String str : ats) {
+                result.add(StringEscapeUtils.unescapeCsv(str.trim()));
+            }
+        }
+        return result;
+    }
+
+    @OnScheduled
+    public void onScheduled(ProcessContext context) {
+        includeCoreAttributes = context.getProperty(INCLUDE_CORE_ATTRIBUTES).asBoolean();
+        coreAttributes = Arrays.stream(CoreAttributes.values()).map(CoreAttributes::key).collect(Collectors.toSet());
+        destinationContent = OUTPUT_OVERWRITE_CONTENT.getValue().equals(context.getProperty(DESTINATION).getValue());
+        nullValForEmptyString = context.getProperty(NULL_VALUE_FOR_EMPTY_STRING).asBoolean();
+     }
+
+    @Override
+    public void onTrigger(ProcessContext context, ProcessSession session) throws ProcessException {
+        final FlowFile original = session.get();
+        if (original == null) {
+            return;
+        }
+        if(context.getProperty(ATTRIBUTES_REGEX).isSet()) {
+            pattern = Pattern.compile(context.getProperty(ATTRIBUTES_REGEX).evaluateAttributeExpressions(original).getValue());
```

Comment:
Local usage of variable only. Should not be defined as field/instance member (same as attributeList).